File: androidApp/src/main/java/com/commuto/interfacemobile/android/p2p/messages/MakerInformationMessage.kt
package com.commuto.interfacemobile.android.p2p.messages

import java.util.*

/**
 * This represents a Maker Information Message, as described in the [Interface Specification](https://github.com/jimmyneutront/commuto-whitepaper/blob/main/commuto-interface-specification.txt).
 *
 * @property swapID The ID of the swap for which maker information is being sent.
 * @property settlementMethodDetails The maker's settlement method details.
 */
data class MakerInformationMessage(val swapID: UUID, val settlementMethodDetails: String)

File: iosApp/iosApp/p2p/messages/MakerInformationMessage.swift
//
//  MakerInformationMessage.swift
//  iosApp
//
//  Created by jimmyt on 8/25/22.
//  Copyright © 2022 orgName. All rights reserved.
//

import Foundation

/**
 This represents a Maker Information Message, as described in the [Interface Specification](https://github.com/jimmyneutront/commuto-whitepaper/blob/main/commuto-interface-specification.txt).
 */
struct MakerInformationMessage {
    /**
     The ID of the swap for which maker information is being sent.
     */
    let swapID: UUID
    /**
     The maker's settlement method details, or `nil` if no such data exists.
     */
    let settlementMethodDetails: String?
}
